lib/msf/core/post/file_stat.rb

```diff
@@ -0,0 +1,169 @@
+# -*- coding: binary -*-
+#
+module Msf::Post::FileStat
+
+class Stat
+  attr_accessor :stathash
+
+ @@ftypes = [
+    'fifo', 'characterSpecial', 'directory',
+    'blockSpecial', 'file', 'link', 'socket'
+  ]
+
+
+  def initialize(filename, session)
+    data = session.shell_command_token("stat --format='%d,%i,%h,%u,%g,%t,%s,%B,%b,%x,%y,%z,%f' #{filename}").to_s.chomp
+    data = data.split(",")
+    @stathash = Hash.new
+    @stathash['st_dev'] = data[0].to_i
+    @stathash['st_ino'] = data[1].to_i
+    @stathash['st_nlink'] = data[2].to_i
+    @stathash['st_uid'] = data[3].to_i
+    @stathash['st_gid'] = data[4].to_i
+    @stathash['st_rdev'] = data[5].to_i
+    @stathash['st_size'] = data[6].to_i
+    @stathash['st_blksize'] = data[7].to_i
+    @stathash['st_blocks'] = data[8].to_i
+    @stathash['st_atime'] = data[9].to_time
+    @stathash['st_mtime'] = data[10].to_time
+    @stathash['st_ctime'] = data[11].to_time
```

Comment:
While this is nicer IMHO, it's unfortunately inconsistent with the Meterpreter `FileStat` which uses flat integers. Can you remove the time conversion and keep them as integers?

Reply:
okay I'll change this to integers :+1: